### PR TITLE
Popup v6 windows

### LINKED ISSUE #371: [Bug] Popup being dismissed when window is resized or loses focus
### Description
Despite the following:
CanBeDismissedByTappingOutsideOfPopup="False"

The popup is still being dismissed when the window is resized or it loses focus.

### Steps to Reproduce

1. Create a popup, set CanBeDismissedByTappingOutsideOfPopup to false
2. Run app on Windows machine
3. Open popup
4. Resize window or click on another window to change focus

### Expected Behavior
Expect the popup to not be dismissed

### Actual Behavior
Popup is dismissed

### Basic Information

- Version with issue: 8.0.0-preview3 

- IDE: VS2022 Preview
- Platform Target Frameworks:
  - UWP:  10.0.19041

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

 

The POC replaces Flyout with Popup. The main reason for that is that popup doesn't dismiss. Flyout is similar to the context menu, while Popup is like a dialog.

 ### Linked Issues ###
 

 - Fixes #371 

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

**Comment by bijington** [comment]
Thanks for this @VladislavAntonyuk it is in my queue to review as part of all open Popup related issues.

**Comment by VladislavAntonyuk** [comment]
@bijington it still needs some changes (mostly related to background color). However, the main issue with disappearing during resizing and minimizing the window is solved.

**Review by brminnick — CHANGES_REQUESTED** [review]
Thanks Vlad! This is looking really good!

But, I'm seeing some unexpected behavior when using an Anchor. In the sample app, the Popup appears far to the right of the anchor.
ScreenFlow

File: src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs [hunk 1]
```diff
@@ -10,17 +15,179 @@ public static class PopupExtensions
...
-		var view = popup.Content;
-		if (view.Background is null && flyout.Control is not null)
+		if (mauiPopup.Child is FrameworkElement content)
 		{
-			flyout.Control.Background = color.ToPlatform();
+			var backgroundProperty = content.GetType().GetProperty("Background");
+			if (backgroundProperty != null)
+			{
+				backgroundProperty.SetValue(content, color.ToPlatform());
+			}
```
Comment (pictos) [inline src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs:31]:
```suggestion
backgroundProperty?.SetValue(content, color.ToPlatform());

```

File: src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs [hunk 2]
```diff
@@ -10,17 +15,179 @@ public static class PopupExtensions
...
+		ArgumentNullException.ThrowIfNull(mauiContext);
+		mauiPopup.PlacementTarget = popup.Anchor?.ToPlatform(mauiContext);
+	}
+
+	/// <summary>
+	/// 
+	/// </summary>
+	/// <param name="platformPopup"></param>
+	/// <param name="popup"></param>
+	/// <param name="mauiContext"></param>
```
Comment (pictos) [inline src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs:52]: a lot of methods don't have the inline docs filled in. Can you address that?

**Review by brminnick — APPROVED** [review]

**Comment by brminnick** [comment]
Thanks Vlad!! I approve. 

I've added the https://github.com/CommunityToolkit/Maui/labels/do%20not%20merge to ensure we don't accidentally merge this PR before releasing `CommunityToolkit.Maui v5.4.0`. 

This PR will be included in our release of `CommunityToolkit.Maui v6.0.0`.

**Review by bijington — APPROVED** [review]
Yes @VladislavAntonyuk !

**Pull request merged**